Possible flakyness in test: ert_tests/ensemble_evaluator/test_sync_ws_duplexer

Error:
```
=================================== FAILURES ===================================
_____________________________ test_immediate_stop ______________________________

unused_tcp_port = 49643, ws = <function ws.<locals>._start_ws at 0x1308743a0>

    def test_immediate_stop(unused_tcp_port, ws):
        # if the duplexer is immediately stopped, it should be well behaved and close
        # the connection normally
        closed = False
    
        async def handler(websocket, path):
            try:
                await websocket.recv()
            except ConnectionClosedOK:
                nonlocal closed
                closed = True
    
        ws("localhost", unused_tcp_port, handler)
    
        duplexer = SyncWebsocketDuplexer(
            f"ws://localhost:{unused_tcp_port}",
            f"ws://localhost:{unused_tcp_port}",
            None,
            None,
        )
    
        duplexer.stop()
    
>       assert closed
E       assert False

ert_tests/ensemble_evaluator/test_sync_ws_duplexer.py:71: AssertionError
```

--- a/tests/ert_tests/ensemble_evaluator/test_sync_ws_duplexer.py
+++ b/tests/ert_tests/ensemble_evaluator/test_sync_ws_duplexer.py
@@ -1,7 +1,7 @@
 import asyncio
 from contextlib import ExitStack
 from http import HTTPStatus
-from threading import Thread
+from threading import Thread, Event
 import time
 from unittest.mock import patch
 
@@ -48,14 +48,13 @@ def ws(event_loop: asyncio.AbstractEventLoop):
 def test_immediate_stop(unused_tcp_port, ws):
     # if the duplexer is immediately stopped, it should be well behaved and close
     # the connection normally
-    closed = False
+    closed = Event()
 
     async def handler(websocket, path):
         try:
             await websocket.recv()
         except ConnectionClosedOK:
-            nonlocal closed
-            closed = True
+            closed.set()
 
     ws("localhost", unused_tcp_port, handler)
 
@@ -68,7 +67,7 @@ def test_immediate_stop(unused_tcp_port, ws):
 
     duplexer.stop()
 
-    assert closed
+    assert closed.wait(10)
 
 
 def test_failed_connection():

